MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs

```diff
@@ -51,8 +51,18 @@ public void UpdateUI(FullTrack a_latestTrack = null)
             if (string.IsNullOrEmpty(artworkURL))
             {
                 Console.WriteLine("No track playing. Getting last track.");
-
-                artworkURL = APIRequestor.Instance.GetLatestTrack().Id;//Get the track ID
+                try
+                {
+                    if (APIRequestor.Instance.GetLatestTrack().Id != null)
+                    {
+                        artworkURL = APIRequestor.Instance.GetLatestTrack().Id;//Get the track ID
+                        APIRequestor.t_id = artworkURL; //Setting the temp id to current one
+                    }
+                }catch(NullReferenceException e)// Internet connection Failure of not getting
+                {
+                    Console.WriteLine(e.StackTrace);
+                    artworkURL = APIRequestor.t_id; //Getting backup of last used id
```

Comment:
I don't like this. Whilst yes we're handling there being an error, we have no guarantee that value is safe, nor do I like us assuming it's not garbage. I'd rather we didn't set anything, and just logged an error here.

Reply:
Okay let me change it to return it null.

Reply:
I fixed it with the last commit